flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/ShardConsumer.java

```diff
@@ -206,7 +206,13 @@ private void deserializeRecordForCollectionAndUpdateState(UserRecord record)
 		final long approxArrivalTimestamp = record.getApproximateArrivalTimestamp().getTime();
 
 		final T value = deserializer.deserialize(
-			keyBytes, dataBytes, subscribedShard.getStreamName(), record.getSequenceNumber(), approxArrivalTimestamp);
+				keyBytes,
+				dataBytes,
+				record.getPartitionKey(),
+				record.getSequenceNumber(),
+				approxArrivalTimestamp,
+				subscribedShard.getStreamName(),
+				subscribedShard.getShard().getShardId());
```

Comment:
Fixed the indention.
I don't think that having a separate final variable for this will improve the performance a lot. The methods are computing anything.